Sources/Swift/Core/Tools/ViewCapture/SentryUIRedactBuilder.swift

```diff
@@ -214,100 +349,141 @@ final class SentryUIRedactBuilder {
         return ObjectIdentifier(containerClass) == redactContainerClassIdentifier
     }
 
+    /// Determines whether a given view should be redacted based on configuration and heuristics.
+    ///
+    /// Order of checks:
+    /// 1. Per‑instance override via `SentryRedactViewHelper.shouldMaskView`.
+    /// 2. Class‑based membership in `redactClassesIdentifiers` (optionally constrained by layer type).
+    /// 3. Special case handling for `UIImageView` (bundle image exemption).
     private func shouldRedact(view: UIView) -> Bool {
+        // First we check if the view instance was marked to be masked
         if SentryRedactViewHelper.shouldMaskView(view) {
             return true
         }
-        if let imageView = view as? UIImageView, containsRedactClass(UIImageView.self) {
+
+        // Extract the view and layer types for checking
+        let viewType = type(of: view)
+        let layerType = type(of: view.layer)
+
+        // Check if the view is supposed to be redacted
+        guard containsRedactClass(viewClass: viewType, layerClass: layerType) else {
+            return false
+        }
+
+        // We need to perform special handling for UIImageView
+        if let imageView = view as? UIImageView {
             return shouldRedact(imageView: imageView)
         }
-        return containsRedactClass(type(of: view))
+
+        return true
     }
     
+    /// Special handling for `UIImageView` to avoid masking tiny gradient strips and
+    /// bundle‑provided assets (e.g. SF Symbols or app assets), which are unlikely to contain PII.
     private func shouldRedact(imageView: UIImageView) -> Bool {
-        // Checking the size is to avoid redact gradient background that
-        // are usually small lines repeating
-        guard let image = imageView.image, image.size.width > 10 && image.size.height > 10  else { return false }
+        // Checking the size is to avoid redacting gradient backgrounds that are usually
+        // implemented as very thin repeating images.
+        guard let image = imageView.image, image.size.width > 10 && image.size.height > 10  else {
```

Comment:
`l`: If you have any clue why we use `10` for the size, please add the info as a comment.